### PR TITLE
Add Xcode 14 build job on CI

### LINKED ISSUE #4073: Add Xcode 14 beta to CircleCI configuration
We should build and test the project in the latest Xcode 14 beta available on CircleCI to detect any regressions related to new Xcode or iOS versions. Some of us are already developing the project in Xcode 14 and on iOS 16, but running these tests on CI would give developers more confidence that we’re ready for the GM release of these versions.

/cc @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by azarovalex**
Fixes #4073.

This PR adds a new build job with Xcode 14 beta, looks like this version has a bug that requires explicit `xcodebuild -resolvePackageDependencies` before building the SPM package. Without it, Xcode [crashes in a strange way](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/9319/workflows/d3eeb68d-2869-4614-b41d-718a7721f819/jobs/83984), I cannot reproduce it locally, only happens on CI.

Also added screenshots for iOS 16 and fixed a new behavior of DistanceFormatter in the Hebrew language.

### CODE REVIEW

**Comment by azarovalex** [comment]
I don't think we will have more screenshots than one folder for each iOS version we support, so it probably won't grow in the near future.

File: Tests/MapboxCoreNavigationTests/DistanceFormatterTests.swift [hunk 1]
```diff
@@ -123,12 +123,24 @@ class DistanceFormatterTests: TestCase {
...
         assertDistance(Measurement(value: 951,      unit: .meters),     displayed: "950 מ׳",     quantity: "950")
-        assertDistance(Measurement(value:   1,      unit: .kilometers), displayed: "1 ק״מ",      quantity: "1")
-        assertDistance(Measurement(value:   1.001,  unit: .kilometers), displayed: "1 ק״מ",      quantity: "1")
-        assertDistance(Measurement(value:   2.5,    unit: .kilometers), displayed: "2.5 ק״מ",    quantity: "2.5")
-        assertDistance(Measurement(value:   2.9,    unit: .kilometers), displayed: "2.9 ק״מ",    quantity: "2.9")
-        assertDistance(Measurement(value:   3,      unit: .kilometers), displayed: "3 ק״מ",      quantity: "3")
-        assertDistance(Measurement(value:   3.5,    unit: .kilometers), displayed: "4 ק״מ",      quantity: "4")
+
+        func iOS16Fix(_ input: String) -> String {
+            // Formatter in iOS 16 now returns U+202F NARROW NO-BREAK SPACE instead of simple space
```
Comment (1ec5) [inline Tests/MapboxCoreNavigationTests/DistanceFormatterTests.swift:128]: Cool, it’s good that we’re able to catch this.

**Review by 1ec5 — APPROVED** [review]

**Comment by MaximAlien** [comment]
> I don't think we will have more screenshots than one folder for each iOS version we support, so it probably won't grow in the near future.

We miss snapshots for iOS 12, and it'd be great to add them. Snapshot tests for this version would probably catch issues like this one: https://github.com/mapbox/mapbox-navigation-ios/issues/4145.

File: .circleci/config.yml [hunk 2]
```diff
@@ -392,6 +391,9 @@ jobs:
             - run:
                 name: Move Xcode project aside
                 command: rm -rf *.xcodeproj
+            - run:
```
Comment (MaximAlien) [inline .circleci/config.yml:394]: Do we still need this for Xcode 14 GA?

**Comment by Udumft** [comment]
>I don't think we will have more screenshots than one folder for each iOS version we support, so it probably won't grow in the near future.

But deleting old snapshots from working copy doesn't remove it from the repo itself. There is a [size limit](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#repository-size-limits) for github repository we should not exceed. Otherwise we'll have to clean git history sometime, which is unpleasant action, especially for public repos. [Git LFS](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-git-large-file-storage) is suggested as a solution for storing various resource files, but it was a bit laggy when I used it last time several years ago.

**Pull request merged**